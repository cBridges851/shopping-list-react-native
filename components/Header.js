import React from "react";
import { View, Text, StyleSheet } from "react-native";

export default class Header extends React.Component {
    render() {
        return (
            <View style={styles.header}>
                <Text style={styles.headerText}>{this.props.title}</Text>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    header: {
        backgroundColor: "#128DFA",
        padding: 20,
        height: 75,
    },
    headerText: {
        color: "#FFFFFF",
        fontSize: 25, 
        textAlign: "center"
    }
});
