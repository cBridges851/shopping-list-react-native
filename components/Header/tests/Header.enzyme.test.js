/**
 * @jest-environment jsdom
 */

import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { mount } from "enzyme";
import { Header } from "../Header";
Enzyme.configure({ adapter: new Adapter() })

describe("Using the Header component", () => {
    let props = {
        title: "Title With a Head"
    };

    let wrapper = mount(<Header title={props.title}/>)

    test("it renders correctly", () => {
        expect(wrapper).toMatchSnapshot()
    });

    describe("given the title 'Title With a Head'", () => {
        test("then it should render this title", () => {
            let header = wrapper.find("Header");
            expect(header.text()).toBe("Title With a Head");
        });
    });

    describe("given the initial props", () => {
        describe("when the props are updated", () => {
            test("then it should render the updated title", () => {
                wrapper.setProps({
                    title: "Title Without a Head"
                });

                let header = wrapper.find("Header");
                expect(header.text()).toBe("Title Without a Head");
            });
        })
    });
});

