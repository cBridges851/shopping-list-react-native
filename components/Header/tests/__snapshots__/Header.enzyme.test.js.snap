// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the Header component it renders correctly 1`] = `ReactWrapper {}`;
